src/core/qgsvectorfilewriter.cpp

```diff
@@ -2831,7 +2831,7 @@ bool QgsVectorFileWriter::deleteShapeFile( const QString &fileName )
   QDir dir = fi.dir();
 
   QStringList filter;
-  const char *suffixes[] = { ".shp", ".shx", ".dbf", ".prj", ".qix", ".qpj" };
+  const char *suffixes[] = { ".shp", ".shx", ".dbf", ".prj", ".qix", ".qpj", ".cpg", ".sbn", ".sbx", ".idm", ".ind" };
```

Comment:
I know it's not introduced by this PR, but I don't think we should not treat the same way the mandatory files (AFAIK ".shp", ".shx", ".dbf") and the optional ones.

I mean that this function is setting `ok` to false even if a single one of these files is missing (and cannot be deleted).

I think you should check for file existence before attempting a remove, and fail only if the missing file is one of the "importants".

Reply:
I've tested QgsVectorFileWriter.deleteShapeFile() in Python console, and it seems deleteShapeFile() returns false only if one of the files is locked and cannot be removed.
If fact it only tries to delete the files that actually are present in the directory of the given shapefile
https://github.com/qgis/QGIS/blob/7410fb0a5238cfdc14707d5cfc65fa4c89a89624/src/core/qgsvectorfilewriter.cpp#L2890-L2898
Maybe I misunderstood?

Reply:
> Maybe I misunderstood?

No, you are correct, I didn't notice the entryList, forget my comment.